libr/core/cmd.c

```diff
@@ -27,32 +27,48 @@
 #include <sys/utsname.h>
 #endif
 
-#define MAX_NUM_CMD_DESCRIPTORS 1024
-RCmdDescriptor cmd_descriptors[MAX_NUM_CMD_DESCRIPTORS];
-int cmd_descriptors_len = 0;
-
-#define REGISTER_CMD_DESCRIPTOR(name)																		\
-	if (cmd_descriptors_len >= R_ARRAY_SIZE (cmd_descriptors)) { 					\
-		eprintf ("Please increase MAX_NUM_CMD_DESCRIPTORS in libr/core/cmd.c\n"); \
-		exit(1);																														\
-	}																																			\
-	cmd_descriptors_len++
-
-#define DEFINE_CMD_DESCRIPTOR(name_)																	\
-	REGISTER_CMD_DESCRIPTOR(name_);																			\
-	cmd_descriptors[cmd_descriptors_len-1].name = #name_;								\
-	cmd_descriptors[cmd_descriptors_len-1].help_msg = help_msg_##name_;
-
-#define DEFINE_CMD_DESCRIPTOR_WITH_DETAIL(name_)												\
-	REGISTER_CMD_DESCRIPTOR(name_);																				\
-	cmd_descriptors[cmd_descriptors_len-1].name = #name_;									\
-	cmd_descriptors[cmd_descriptors_len-1].help_msg = help_msg_##name_;		\
-	cmd_descriptors[cmd_descriptors_len-1].help_detail = help_detail_##name_;
-
-#define DEFINE_CMD_DESCRIPTOR_SPECIAL(cmd, name_)                          \
-	REGISTER_CMD_DESCRIPTOR(name_);                                     \
-	cmd_descriptors[cmd_descriptors_len-1].name = #cmd;                 \
-	cmd_descriptors[cmd_descriptors_len-1].help_msg = help_msg_##name_; \
+#define DEFINE_CMD_DESCRIPTOR(core, name_)				\
+	{																								\
+		RCmdDescriptor *d = R_NEW0 (RCmdDescriptor);	\
+		if (d) {																			\
+			d->name = #name_;														\
+			d->help_msg = help_msg_##name_;							\
+			r_list_append (core->cmd_descriptors, d);		\
+		}																							\
+	}
+
+#define DEFINE_CMD_DESCRIPTOR_WITH_DETAIL(core, name_)	\
+	{																											\
+		RCmdDescriptor *d = R_NEW0 (RCmdDescriptor);				\
+		if (d) {																						\
+			d->name = #name_;																	\
+			d->help_msg = help_msg_##name_;										\
+			d->help_detail = help_detail_##name_;							\
+			r_list_append (core->cmd_descriptors, d);					\
+		}																										\
+	}
+
+#define DEFINE_CMD_DESCRIPTOR_WITH_DETAIL2(core, name_)	\
```

Comment:
backslash should  be only 1 char after the last char of the line, not at random places